### PR TITLE
added a pissibility to manage edit profile requests

### PR DESCRIPTION
**Pull request by serhiidotsyk**

### CODE REVIEW

**Review by SergiyKozachok — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/RequestService.cs [hunk 1]
```diff
@@ -141,5 +142,120 @@ public async Task<(bool, string)> HandlePatientToDoctorRequest(int id, bool isAp
...
+                .Include(u => u.User.Passports)
+                .Include(u => u.User.Diplomas)
+                .Include(u => u.User.Licenses)
+                .GroupBy(tu => tu.UserId)
+                .Select(tu => tu.LastOrDefault())
+                .AsEnumerable();
+
+            var amount = uniqueTmpUsers.Count();
+            var usersRequest = PaginationHelper<TemporaryUser>.GetPageValues(uniqueTmpUsers.AsQueryable(), pageConfig.PageSize, pageConfig.PageIndex);
+            var UsersEditProfile = usersRequest.AsEnumerable()
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/RequestService.cs:165]: Why no mapper here

File: InternetHospital.BusinessLogic/Services/RequestService.cs [hunk 2]
```diff
@@ -141,5 +142,120 @@ public async Task<(bool, string)> HandlePatientToDoctorRequest(int id, bool isAp
...
+                return null;
+            }
+
+            var uniqueTmpUsers = tmpUsers
+                .Include(u => u.User.Passports)
+                .Include(u => u.User.Diplomas)
+                .Include(u => u.User.Licenses)
+                .GroupBy(tu => tu.UserId)
+                .Select(tu => tu.LastOrDefault())
+                .AsEnumerable();
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/RequestService.cs:161]: As i remember, select should return ienumerable

File: InternetHospital.BusinessLogic/Services/RequestService.cs [hunk 3]
```diff
@@ -141,5 +142,120 @@ public async Task<(bool, string)> HandlePatientToDoctorRequest(int id, bool isAp
...
+                if (tmpUser.User.Doctor == null)
+                {
+                    Mapper.Map<TemporaryUser, User>(tmpUser, user,
+                        config => config.AfterMap((src, dest) =>
+                        {
+                            dest.StatusId = 3;
+                        }));
+
+                    user.LastStatusChangeTime = DateTime.Now;
+                    _context.Users.Update(user);
```
Comment (TREN-D) [inline InternetHospital.BusinessLogic/Services/RequestService.cs:219]: Better save date in utc

**Review by TREN-D — APPROVED** [review]

**Review by martyniukroman — APPROVED** [review]

File: InternetHospital.BusinessLogic/Services/RequestService.cs (hunk 1 above)
Comment (voznovych) [reply InternetHospital.BusinessLogic/Services/RequestService.cs:165]: Why AsEnumerable?

**Pull request merged**